src/main/java/org/javarosa/core/model/instance/ExternalDataInstance.java

```diff
@@ -38,27 +44,36 @@ private ExternalDataInstance(TreeElement root, String instanceId, String path) {
     }
 
     /**
-     * Builds an ExternalDataInstance
+     * Builds an ExternalDataInstance by parsing the file at the given location or fills in a placeholder if the file
+     * can't be found or derived.
+     *
+     * The placeholder makes it possible to successfully parse a form without an external
+     * instance in cases where a client isn't providing a form-filling interface (e.g. form validation or discovery).
      *
      * @param instanceSrc the value of the instance’s src attribute, e.g., jr://file/…
      * @param instanceId  the ID of the new instance
      * @return a new ExternalDataInstance
-     * @throws IOException                       if FileInputStream can’t find the file, or ElementParser can’t read the stream
-     * @throws InvalidReferenceException         if the ReferenceManager in getPath(String srcLocation) can’t derive a reference
+     * @throws IOException                       if ElementParser can’t read the stream
      * @throws UnfullfilledRequirementsException thrown by {@link TreeElementParser#parse()}
      * @throws XmlPullParserException            thrown by {@link TreeElementParser#parse()}
      * @throws InvalidStructureException         thrown by {@link TreeElementParser#parse()}
      */
     public static ExternalDataInstance build(String instanceSrc, String instanceId)
-        throws IOException, UnfullfilledRequirementsException, XmlPullParserException, InvalidStructureException, InvalidReferenceException {
-        TreeElement root = parseExternalInstance(instanceSrc, instanceId);
+        throws IOException, UnfullfilledRequirementsException, XmlPullParserException, InvalidStructureException {
+        TreeElement root;
+        try {
+            root = parseExternalInstance(instanceSrc, instanceId);
+        } catch (FileNotFoundException | InvalidReferenceException e) {
+            logger.info("External instance not found, falling back to placeholder");
```

Comment:
I decided to leave this in because the other logging around form load is often helpful from Collect.